core/src/main/scala/com/raphtory/algorithms/generic/filters/VertexQuantileFilter.scala

```diff
@@ -0,0 +1,136 @@
+package com.raphtory.algorithms.generic.filters
+
+import com.raphtory.algorithms.api.Bounded
+import com.raphtory.algorithms.api.GraphPerspective
+import com.raphtory.algorithms.api.Identity
+import com.raphtory.algorithms.generic.NodeList
+
+import scala.collection.Searching.Found
+import scala.collection.Searching.InsertionPoint
+import scala.language.implicitConversions
+import scala.math.Numeric.Implicits.infixNumericOps
+import scala.reflect.ClassTag
+
+/**
+  * {s}`EdqeQuantileFilter()`
+  * : Filtered view of the graph based on edge weight
+  *
+  *  This creates a filtered view of the graph where nodes are removed based on where their value of a given property/state lies in the global
+  *  distribution of vertex states, understood in this algorithm as percentiles. For example, one can create a view of the
+  *  graph including only the top 50% of vertices in terms of (say) degree. This algorithm does not return any output and is best used in composition
+  *  with other algorithms, using the Chain API. The algorithm is agnostic to whether the network is directed or undirected.
+  *
+  * ## Parameters
+  *
+  *  {s}`lower: Float = 0.0f`
+  *  : The lower cutoff percentile below which nodes will be removed, a float with 0 <= lower <= upper <=1.0 with default value 0.0f
+  *
+  *  {s}`upper: Float = 1.0f`
+  *  : The upper cutoff percentile above which nodes will be removed, a float with 0 <= lower <= upper <=1.0 with default value 1.0f
+  *
+  *  {s}`weightString: String = "weight"`
+  *  : String name of the property/state, defaulting to "weight". As with other weighted algorithms in Raphtory, if no weight property
+  *  is there but multi-edges are present, the number of occurrences of each edge is treated as the weight.
+  *
+  *  {s}`lowerExclusive: Boolean = "false"`
+  *  : whether the inequality on the threshold is strict or not at the lower end
+  *
+  *  {s}`upperExclusive: Boolean = "false"`
+  *  : whether the inequality on the= threshold is strict or not at the upper end
+  *
+  *  {s}`noBins: Int = 1000`
```

Comment:
I was thinking that having a fixed size that does not scale up with number of edges/vertices might be preferable since it will be broadcast across partitions. Also if it's an integer valued property like degree then having more than (maxVal - minVal) bins wouldn't bring any improvement to the quantile precision anyway. But I do agree that maybe a more clever default generation of bins could be helpful, maybe even log spaced ones aimed at power law distributed properties, so will have a think :)